src/Sylius/Bundle/AdminBundle/Resources/config/services.xml

```diff
@@ -25,6 +25,11 @@
             <tag name="sylius.context.locale" priority="128" />
         </service>
 
+        <service id="sylius.security.shop_user_impersonator" class="Sylius\Bundle\CoreBundle\Security\UserImpersonator" public="false">
+            <argument type="service" id="session" />
+            <argument type="string">%sylius_shop.firewall_context_name%</argument>
```

Comment:
It is already coupled, as currently AdminBundle uses service from CoreBundle which uses parameter from ShopBundle, this PR just gets CoreBundle out of the equation :D